src/Orchard.Specs/Bindings/Settings.cs

```diff
@@ -30,6 +22,11 @@ public class Settings : BindingBase {
                     }
 
                     orchardServices.WorkContext.CurrentSite.SiteCulture = cultureName;
+
+                    // Restarting the shell to reset the cache, because the cache entry storing the list of available
+                    // cultures isn't invalidated by the signal in DefaultCultureManager.ListCultures when running
+                    // inside the test webhost.
+                    MvcApplication.RestartTenant("Default");
```

Comment:
This is what I mentioned in my comment under the issue: https://github.com/OrchardCMS/Orchard/issues/8686#issuecomment-1562516639
IMO the workaround is harmless (although not ideal), but changing the culture settings works fine otherwise.